src/runtime/entitlements-manager.js

```diff
@@ -121,23 +123,25 @@ export class EntitlementsManager {
    */
   fetchEntitlementsWithCaching_() {
     return this.storage_.get(ENTS_STORAGE_KEY).then(raw => {
-      // Try cache first.
-      if (raw) {
-        const cached = this.getValidJwtEntitlements_(
-            raw, /* requireNonExpired */ true);
-        if (cached && cached.enablesThis()) {
-          // Already have a positive response.
-          this.positiveRetries_ = 0;
-          return cached;
+      return this.storage_.get(IS_READY_TO_PAY_STORAGE_KEY).then(irtp => {
+        // Try cache first.
+        if (raw) {
+          const cached = this.getValidJwtEntitlements_(
+              raw, /* requireNonExpired */ true, !!irtp);
```

Comment:
Will `!!` actually give us what we want? What types can storage save/return? Aren't they all converted to strings?

Reply:
fixed